main.go

```diff
@@ -339,3 +351,13 @@ func initKubeConfig(kcLocation string) *rest.Config {
 
 	return kubeConfig
 }
+
+func getOutboundIP(upstreamHost string) (string, error) {
```

Comment:
I personally think this is a bit odd, can we add a comment why this is the way we are retrieving the address? I'm not concerned with the expensiveness anymore as we're just doing it once at startup, but reading this in 3 months I'll want to understand why this is the best way of getting the preferred local IP.